### PR TITLE
Feature remove global config in the search/browser engine

### PR DESCRIPTION
**Pull request by shenchucheng**
**Features**
- Replace `parse_obj_as` with `TypeAdapter.validate_python`
- Remove the global config in the search/browser engine
- Allow access to private_context when validating Action using the model_validator(mode="after") decorator
- Replace the `__init__` method with `model_validator` in the `WebBrowserEngine`/`SearchEngine`/`ContextMixin` class to make it more annotation-friendly

**Result**

image

### CODE REVIEW

File: metagpt/configs/browser_config.py [hunk 1]
```diff
@@ -15,6 +15,6 @@ class BrowserConfig(YamlModel):
     """Config for Browser"""
 
     engine: WebBrowserEngineType = WebBrowserEngineType.PLAYWRIGHT
-    browser: Literal["chrome", "firefox", "edge", "ie"] = "chrome"
-    driver: Literal["chromium", "firefox", "webkit"] = "chromium"
-    path: str = ""
+    browser_type: Literal["chromium", "firefox", "webkit", "chrome", "firefox", "edge", "ie"] = "chromium"
```
Comment (geekan) [inline metagpt/configs/browser_config.py:18]: update related yaml

Comment (shenchucheng) [reply metagpt/configs/browser_config.py:18]: image

**Comment by geekan** [comment]
update https://github.com/geekan/MetaGPT-docs/ is also needed.

**Review by geekan — APPROVED** [review]
LGTM

**Pull request merged**